src/Microsoft.AspNetCore.Mvc.Core/ControllerAttribute.cs

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Mvc
+{
+    /// <summary>
+    /// Indicates that the type and any derived types that this attribute is applied to
+    /// are considered a controller by the default controller discovery mechanism, unless
+    /// <see cref="NonControllerAttribute"/> is applied to any type in the hierarchy.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class, AllowMultiple = false, Inherited = true)]
+    public class ControllerAttribute : Attribute
+    {
+    }
```

Comment:
I think that is a separate feature from what we are doing here, can we do it as a later as a separate change? I'm not sure of all the implications that setting the controller name through an attribute the is inherited might have and it's definitely something that we can tackle as a separate issue.